### PR TITLE
[risk=no][RW-8814] Update users free tier usage in multiple threads

### PR DESCRIPTION
**Pull request by tarekmamdouh**
Description:

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have added explanatory comments where the logic is not obvious
- [ ] I have run and tested this change locally, and my testing process is described here
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later
- [ ] If this includes an API change, I have run the E2E tests on this change against my local server with [yarn test-local](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples) because this PR won't be covered by the CircleCI tests 
- [ ] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P)
- [ ] If this change impacts deployment safety (e.g. removing/altering APIs which are in use) I have documented these in the description
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and updated the appropriate API consumers
  * AoU UI
  * [Perf tests](https://github.com/broadinstitute/mcnulty/blob/develop/src/test/scala/services/AoU.scala)
  * [Researcher Directory export](https://github.com/all-of-us/workbench/wiki/Researcher-Directory-(RDR-export))
  * Cron tasks - for Offline*Controllers
  * SumoLogic - for EgressAlert

### CODE REVIEW

File: api/src/main/java/org/pmiops/workbench/billing/FreeTierBillingService.java [hunk 1]
```diff
@@ -298,11 +298,16 @@ private void sendAlertsForCostThresholds(
...
+    Iterable<DbUser> userByUserIdIn =
+        userDao.findAllById(userCosts.keySet().stream().collect(Collectors.toList()));
+    Map<Long, DbUser> usersCache = new HashMap<>();
+    userByUserIdIn.iterator().forEachRemaining(u -> usersCache.put(u.getUserId(), u));
+
     userCosts.forEach(
         (userId, currentCost) -> {
           final double previousCost = previousUserCosts.getOrDefault(userId, 0.0);
           maybeAlertOnCostThresholds(
-              userDao.findUserByUserId(userId),
```
Comment (tarekmamdouh) [inline api/src/main/java/org/pmiops/workbench/billing/FreeTierBillingService.java:305]: This is very non-optimal and causes a lot of delay in the job. I'm batching the query to make it faster.

**Comment by tarekmamdouh** [comment]
I believe this is the main root cause of the delay, I think the fix will help but will still look at the long term solution by creating a task queue.

**Review by yonghaoy — APPROVED** [review]
LG.
Did you poll number from log and see if that can solve this issue? Like how fast this new query would be, and how many user left to be processed when it reached to 10min DDL? 

Also, I will create a ticket to split this cron job to smaller batchs (and will put in next sprint if we agree that has higher priority)

nit: Add security risk in PR title before merging.

**Comment by tarekmamdouh** [comment]
@yonghaoy in the logs the time between the log line preceding this method and the one after it is on average 15s, currently I saw that we reach 30 iterations of 200 users each, if we save the 15s, and say the new query would take 2-3 seconds, then we're saving about 5 mins, which should be sufficient.

File: api/src/main/java/org/pmiops/workbench/billing/FreeTierBillingService.java (hunk 1 above)
Comment (jmthibault79) [reply api/src/main/java/org/pmiops/workbench/billing/FreeTierBillingService.java:305]: this is very similar to the ToS email problem I had.  good update.

**Pull request merged**